### ISSUE TITLE
AttributeError: partially initialized module 'cv2' has no attribute 'VideoCapture'

### ISSUE THREAD

**Issue by emanuelsc3s** [pyscripter/python4delphi#303]
I have the error when running:

used Delphi 10.4.2 and Python 3.9.2 64bit

Traceback (most recent call last):
  File "<string>", line 51, in remover_frame_a_frame_e_fazer_predicoes_e_plotar
AttributeError: partially initialized module 'cv2' has no attribute 'VideoCapture' (most likely due to a circular import)

Help please...

```
def retirar_frame_a_frame_e_fazer_predicoes_e_plotar(path_video, y_true):

  violent_count = 0
  non_violent_count = 0
  contador = 0
  
  vetor_probs = []
  vetor_labels = []
  vetor_imagens = []
  
  cap = cv2.VideoCapture(path_video)

  while (cap.isOpened()):
    ret, frame = cap.read()

    if (ret == False):
      break
    
    img = cv2.cvtColor(frame, cv2.COLOR_BGR2RGB) 

    img_pil = Image.fromarray(img)
    vetor_imagens.append(img_pil)

    img_tensor = transformer(img_pil)
    
    img_tensor.unsqueeze_(0)

    resultado = model.forward(img_tensor)

    probs = torch.exp(resultado)

    probs = probs.max(1)

    probabilidade = probs[0].detach().numpy()[0]
    vetor_probs.append(probabilidade)

    label_predita = probs[1].detach().numpy()[0]
    vetor_labels.append(label_predita)

    contador += 1

    if (label_predita == 1):
      violent_count += 1
    else:
      non_violent_count += 1
    
    taxa_frames_violentos = 100*violent_count/contador
    taxa_frames_nao_violentos = 100*non_violent_count/contador
  
  plt.figure(figsize=(15, 10))
  for k in range(9):
    plt.subplot(3, 3, k+1)
    n = random.randint(0, len(vetor_probs)-1)
    img = vetor_imagens[n]
    label = vetor_labels[n]
    plt.imshow(img)
    plt.xticks([])
    plt.yticks([])
    plt.title('Real: {0}\nPredito: {1}'.format(vetor_classes[y_true], vetor_classes[label]))
  
  plt.show()

  return np.array(vetor_probs), np.array(vetor_labels), taxa_frames_violentos, taxa_frames_nao_violentos
```

**Comment by emanuelsc3s** [comment]
I used

```
import cv2
import pathlib
import os
import numpy as np
import matplotlib.pyplot as plt
from PIL import Image, ImageDraw
import random
import shutil
import zipfile
from tqdm import tqdm

import torch
import torchvision
from torch.utils.data import DataLoader, Dataset
from torchvision.datasets import ImageFolder
from torchvision import transforms
import random
from time import time
from torch.utils.data import random_split
```

**Comment by pyscripter** [comment]
Why do you believe this is P4D issue?

**Comment by alperyazgan** [comment]
Definitely not :-)

I think this fellow friend [ emanualsc3s ] who is asking this question is very young and at the beginning of his/her learning curve.

But meanwhile, If I do not disturb you, I have a question which is, I believe, related with the python4delphi, is there a way to get the standard output (eg. from print()) of running script in the attached (py)thread. IOController could not capture the output while the engine is running with a thread. 

Currently I am using another synced output routine to get the output into VCL.      

Any comment will be helpful, 

Kind regards,
 
O. Alper YAZGAN

**Comment by pyscripter** [comment]
@alperyazgan Please address your question  to the support forum (with some more details).

**Comment by emanuelsc3s** [comment]
> Por que você acredita que este é um problema de P4D?

because I ran the same code in Visual Studio Code and it worked

**Comment by emanuelsc3s** [comment]
> Why do you believe this is P4D issue?

the error happens in the function call:

`vetor_probs, vetor_labels, taxa_frames_violentos, taxa_frames_nao_violentos = retirar_frame_a_frame_e_fazer_predicoes_e_plotar(r'C:\cam2\11.mp4', 1)`

**Comment by pyscripter** [comment]
> because I ran the same code in Visual Studio Code and it worked

This does not show that it is P4D issue.  VS Code may be running a different version of Python with a different sys.path and different versions of the library.

**Comment by SwiftExpat** [comment]
I solved a smiliar situation:

I used demo01 to call each of my imports individually to eventually trace it down to C++ runtime not being installed.  For me the offending package was skimage.  It is possible to see what is missing when you trace the dll load with sysinternals procmon.

I can best illustrate it as:
VS Code --> C++ [python.exe] --> loads python.dll --> imports skimage --> requires C++ runtime (provided by python)

	  Delphi[Demo01.exe] --> loads python.dll --> imports skimage --> requires C++ runtime (provided by OS in path)

So when it runs in VS Code,  the runtime requirement is fulfilled. (at least my best guess)
